File: backend/app/models/exercise.py
import uuid
from datetime import date, datetime

from sqlalchemy import Date, DateTime, ForeignKey, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base


class Exercise(Base):
    __tablename__ = "exercises"

    id: Mapped[str] = mapped_column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    name: Mapped[str] = mapped_column(String(200))
    start_date: Mapped[date] = mapped_column(Date)
    end_date: Mapped[date] = mapped_column(Date)
    location: Mapped[str | None] = mapped_column(String(200), nullable=True)
    # status: active | closed
    status: Mapped[str] = mapped_column(String(20), default="active")
    created_by: Mapped[str] = mapped_column(String(36), ForeignKey("users.id"))
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())

File: frontend/src/types/exercise.ts
export interface Exercise {
  id: string
  name: string
  start_date: string  // ISO date string YYYY-MM-DD
  end_date: string
  location: string | null
  status: 'active' | 'closed'
  created_by: string
  created_at: string
}

export interface ExerciseCreate {
  name: string
  start_date: string
  end_date: string
  location?: string | null
}
